### DISCUSSION TITLE
Single quotes char in Export-Csv and [-UseQuotes AsNeeded] parameter

### DISCUSSION THREAD

**Discussion by kmieciu2003** [PowerShell/PowerShell#19172] [Q&A]
Hello everyone,
I have two questions that may become issues.
First:
Is there any possibility to force Export-Csv function to make single quote char (') instead of double quotes (")?
Second:
Can I add custom char to handle in -UseQuotes AsNeeded option. As far I know, right now char set which trigger quotes is limited to:
-defined delimiter
-new line char
-doube quotes
But I need to add to this list also backslash (\\).

Thank you in advance for your answer and if I should create new issue for that please also let me know.

**Comment by mklement0** [comment]
Speaking in no official capacity whatsoever: My guess is that you're much more likely to receive help with a question of this nature on [Stack Overflow](https://stackoverflow.com/).

* No, you cannot currently make `Export-Csv` use single quotes.
* No, you cannot currently control what triggers quoting of fields (however, you can pass a hard-coded list of fields that should be quoted to `-QuoteFields`)

CSV is a well-defined format, and it calls for _double_ quotes, and `\` has no special meaning, _except_ potentially to escape `"` chars., though `""`-escaping is more widely supported.

So I think what you need calls for a _custom_ solution; e.g. (uses hard-coded delimiter (separator) `,`):

```powershell
$firstRow = $true
[pscustomobject] @{ col1 = 'simple'; col2 = 'a,b'; col3 = 'a\b'; col4 = "a'b"; col5 = "a`nb"} | 
  ForEach-Object -Begin { 
    function convertTo-CsvRow([object[]] $values) {
      $values.ForEach({
          if ($_ -match '[,\\''\r\n]') {
            "'{0}'" -f ($_ -replace '(?<!\\)(\\)*''', '$1$1\''') # escape ' as needed
          }
          else {
            $_
          }
        }) -join ','
    }
  } -Process {
    if ($firstRow) { convertTo-CsvRow $_.psobject.Properties.Name; $firstRow = $false }
    convertTo-CsvRow $_.psobject.Properties.Value
  }
```

Output:

```
col1,col2,col3,col4,col5
simple,'a,b','a\b','a\'b','a
b'
```

---

That said, generally speaking, if you think that what you're looking for would benefit others too, you could create a issue of type "Feature Request / Idea" at https://github.com/PowerShell/PowerShell/issues/new/choose, and make a specific proposal.
